tests/integration/test_charm.py

```diff
@@ -176,13 +180,96 @@ async def test_monitor_user(ops_test: OpsTest) -> None:
     ]
     hosts = ",".join(replica_set_hosts)
     replica_set_uri = f"mongodb://monitor:{password}@{hosts}/admin?replicaSet=mongodb"
-
     admin_mongod_cmd = f"charmed-mongodb.mongo '{replica_set_uri}'  --eval 'rs.conf()'"
     check_monitor_cmd = f"exec --unit {unit.name} -- {admin_mongod_cmd}"
     return_code, _, _ = await ops_test.juju(*check_monitor_cmd.split())
     assert return_code == 0, "command rs.conf() on monitor user does not work"
 
 
+async def test_only_leader_can_set_while_all_can_read_password_secret(ops_test: OpsTest) -> None:
+    """Test verifies that only the leader can set a password, while all units can read it."""
+    # Setting existing password
+    leader_id = await get_leader_id(ops_test)
+    non_leaders = list(UNIT_IDS)
+    non_leaders.remove(leader_id)
+
+    password = "blablabla"
+    await set_password(ops_test, unit_id=non_leaders[0], username="monitor", password=password)
+    password1 = await get_password(ops_test, username="monitor")
+    assert password1 != password
+
+    await set_password(ops_test, unit_id=leader_id, username="monitor", password=password)
+    for _ in UNIT_IDS:
+        password2 = await get_password(ops_test, username="monitor")
+        assert password2 == password
+
+
+@pytest.mark.usefixtures("only_with_juju_secrets")
+async def test_reset_and_get_password_secret_same_as_cli(ops_test: OpsTest) -> None:
+    """Test verifies that we can set and retrieve the correct password using Juju 3.x secrets."""
+    new_password = str(uuid4())
+
+    # Resetting existing password
+    leader_id = await get_leader_id(ops_test)
+    result = await set_password(
+        ops_test, unit_id=leader_id, username="monitor", password=new_password
+    )
+
+    secret_id = result["secret-id"].split("/")[-1]
+
+    # Getting back the pw programmatically
+    password = await get_password(ops_test, username="monitor")
+
+    # Getting back the pw from juju CLI
+    complete_command = f"show-secret {secret_id} --reveal --format=json"
+    _, stdout, _ = await ops_test.juju(*complete_command.split())
+    data = json.loads(stdout)
+
+    assert password == new_password
+    assert data[secret_id]["content"]["Data"]["monitor-password"] == password
+
+
+@pytest.mark.usefixtures("only_without_juju_secrets")
+async def test_reset_and_get_password_no_secret(ops_test: OpsTest, mocker) -> None:
+    """Test verifies that we can set and retrieve the correct password using Juju 2.x."""
```

Comment:
in tox this test is set up with `3.2.0.1`, I don't think this test runs on CI with `2.x`. i.e. this test never runs on CI, this is okay with me for now. But maybe in the future we can run it on CI with juju `2.x`?

Reply:
We not supporting Juju 2.x anymore

Reply:
...but as long as we did...

I was running these tests on both Juju2 and Juju3 pipelines. 
The ones with the `only_with_juju_secrets` fixture were ONLY executed on Juju3 and skipped on Juju2 while for the `only_without_juju_secrets` it was the opposite.

Thus this particular test was only run on Juju2.

NOW, if there's literally no more Juju3 in pipelines, then all marked tests (`only_without_juju_secrets`) can be removed, together with both fixtures themselves.